run_unittests.py

```diff
@@ -3587,6 +3587,19 @@ def test_apple_bitcode_modules(self):
         self.build()
         self.run_tests()
 
+    def test_deterministic_dep_order(self):
+        '''
+        Test that the dependencies are always listed in a deterministic order.
+        '''
+        testdir = os.path.join(self.common_test_dir, '206 dep order')
+        self.init(testdir)
+        with open(os.path.join(self.builddir, 'build.ninja')) as bfile:
+            for line in bfile:
+                if 'build myexe:' in line or 'build myexe.exe:' in line:
+                    self.assertIn('liblib1.a liblib2.a', line)
+                    return
+        raise RuntimeError('Could not find the build rule')
```

Comment:
Instead of doing this, you probably want to cover the test function with a decorator like: `@unittest.skipIf(self.backend is not Backend.ninja)`. This test is run on all platforms.

Reply:
This is actually part of LinuxlikeTests. and other tests in this class (test_order_of_l_arguments for example) already assume that ninja is being used. Is that assumption incorrect?